src/runtime/storageNG/unified-store.ts

```diff
@@ -44,10 +45,25 @@ export abstract class UnifiedStore implements Comparable<UnifiedStore> {
   // JSON representation for insertion into the serialization.
   // tslint:disable-next-line no-any
   abstract toLiteral(): Promise<any>;
-  abstract cloneFrom(store: UnifiedStore): void;
-  abstract modelForSynchronization(): {};
-  abstract on(type: string, fn: Consumer<{}>, target: {}): void;
+
+  // TODO: These properties/methods do not belong on UnifiedStore. They should
+  // probably go on some other abstraction like UnifiedActiveStore or similar.
+  abstract source?: string;
   abstract description: string;
+  cloneFrom(store: UnifiedStore): void {}
+  async modelForSynchronization(): Promise<{}> {
+    return await this.toLiteral();
+  }
+  on(type: string, fn: Consumer<{}>, target: {}): void {}
+
+  /**
+   * Hack to cast this UnifiedStore to the old-style class StorageStub.
+   * TODO: Fix all usages of this method to handle new-style stores, and then
+   * delete.
+   */
+  castToStorageStub(): StorageStub {
+    return this as unknown as StorageStub;
```

Comment:
you could check  (this instanceof StorageStub), and if not throw an exception?

Reply:
Good idea, done!

Reply:
Actually I couldn't do this, it caused a circular dependency between UnifiedStore and StorageStub. I've added tags for all the subclasses, and assert based on that. It's almost as good...